nixos/modules/system/boot/initrd-ssh.nix

```diff
@@ -87,13 +87,32 @@ in
       '';
     };
 
-    authorizedKeys = mkOption {
-      type = types.listOf types.str;
-      default = config.users.users.root.openssh.authorizedKeys.keys;
-      defaultText = literalExpression "config.users.users.root.openssh.authorizedKeys.keys";
-      description = lib.mdDoc ''
-        Authorized keys for the root user on initrd.
-      '';
+    authorizedKeys = {
+
+      keys = mkOption {
```

Comment:
I don't really like the stutter because of the twice keys. I would suggest to keep authorizedKeys and name the new option authorizedKeyFiles and also keep the location of the options.